nodes/teleop_legged_keyboard.py

```diff
@@ -142,17 +143,17 @@ def wait_for_subscribers(self):
         i = 0
         while not rospy.is_shutdown() and (self.twist_publisher.get_num_connections() == 0 or
                                            self.pose_publisher.get_num_connections() == 0):
-            if i == 4 and self.twist_publisher.get_num_connections() == 0 \
-                    and self.pose_publisher.get_num_connections() == 0:
-                print("Waiting for subscriber to connect to {}".format(self.twist_publisher.name))
-                print("Waiting for subscriber to connect to {}".format(self.pose_publisher.name))
-            elif i == 4 and self.twist_publisher.get_num_connections() == 0:
-                print("Waiting for subscriber to connect to {}".format(self.twist_publisher.name))
-            elif i == 4 and self.pose_publisher.get_num_connections() == 0:
-                print("Waiting for subscriber to connect to {}".format(self.pose_publisher.name))
+            if i == self.delay_wait_print:
+                if self.twist_publisher.get_num_connections() == 0 and self.pose_publisher.get_num_connections() == 0:
```

Comment:
As for me it is better to use Python style and don't use additional if's, With next code, you can cover all statements
```
rospy.loginfo("Waiting for subscriber to connect to {}".format(self.twist_publisher.name)) if self.twist_publisher.get_num_connections() == 0 else None

rospy.loginfo("Waiting for subscriber to connect to {}".format(self.pose_publisher.name)) if self.pose_publisher.get_num_connections() == 0 else None
```
or
```
if self.twist_publisher.get_num_connections() == 0:
rospy.loginfo("Waiting for subscriber to connect to {}".format(self.twist_publisher.name))

if self.pose_publisher.get_num_connections() == 0:
rospy.loginfo("Waiting for subscriber to connect to {}".format(self.pose_publisher.name))

```

Reply:
Thanks, I got it now.